src/composables/addPost.js

```diff
@@ -0,0 +1,98 @@
+import { reactive, ref } from "vue";
+import slugify from "slugify";
+import { authenticate } from "@/utils/authenticate";
+import { useRegister } from "./register";
+import { usePostStore } from "@/stores/postStore";
+import router from "@/router";
+
+export const useAddPost = () => {
+  const { isErrorPresent } = useRegister();
+  const { clearData } = useRegister();
+  const { handleAddPost } = usePostStore();
+  const postDetails = reactive({
+    title: null,
+    slug: null,
+    description: "",
+    postImage: null,
+    imageUrl: null,
+  });
+
+  const errorMessage = reactive({
+    title: null,
+    description: null,
+    postImage: null,
+  });
+
+  const validateForm = ref(false);
+  const showConfirmationModal = ref(false);
+
+  const closeConfirmationModal = () => {
+    showConfirmationModal.value = false;
+    router.push("/post-list");
+  };
+  const uploadImage = (e) => {
```

Comment:
according to function code it not looks like function is uploading any image @aayushbatra02